content/blog/2015-06-03-service-discovery-introduction.md

```diff
@@ -0,0 +1,328 @@
+---
+title: Service Discovery Introduction
+created_at: 2015-06-02
+kind: article
+author: Fabian Reinartz
+---
+
+This week we released Prometheus v0.14.0 — a version with many long-awaited additions
+and improvements.
+
+On the user side, new mechanisms for service discovery aim to make it easier to
+dynamically update the targets Prometheus scrapes. In addition to DNS-SRV records,
+[Consul](https://www.consul.io) is now supported out-of-the-box and a file-based interface
+allows you to connect your own discovery mechanism.
+Over time, we plan to extend Prometheus with native support for other common ways of doing
+service discovery.
+
+Aside from many smaller fixes and improvements, you can now also reload your configuration at 
+runtime by sending a `SIGHUP` to the Prometheus process. For a full list of changes, check the
+[changelog for this release](tba).
+
+In this blog post, we will take a closer look at the built-in service discovery mechanisms and provide
+small practical examples. As an additional resource
+[Prometheus's configuration documentation](/docs/operating/configuration),
+can be helpful.
+
+
+## Prometheus and targets
+
+For proper understanding of this blog post, we have to take a look at how Prometheus sees targets.
+To Prometheus each scrape target is nothing more but another set of labels – even its address, which
+is defined by the `__address__` label.
+
+In configurations we have a hierarchie of labels. There are global labels, which are assigned to
+every target scraped by the Prometheus instance. One level above is the `job` label, that is defined directly
+in the configuration file. On the next level are labels defined in the target group from which
+a label is derived. Finally, there are labels that are set per-target as we will see soon.
+
+Each layer of labels overwrites any colliding labels from the layers below. Eventually, we have a flat
+set of labels that describe a single target. Those labels are then attached to every metric that
+is scraped from this target.
+
+
+## Scrape configurations and relabeling
+
+Aside from switching from the ASCII protocol buffer format to YAML, a fundamental change to 
+Prometheus's configuration is the switch from per-job to scrape configurations.
+For simple setups the two can be thought of as almost equivalent.
+At the same time, scrape configurations allow greater flexibility for more advanced use cases.
+
+Each scrape configuration is assigned a job name which serves as a default value for the
+`job` label. The `job` label can then be redefined for whole target groups or single targets.
+For example, we can define two target groups, each of which defines targets for one job. 
+To scrape them with the same parameters, we can configure them as follows:
+
+```
+scrape_configs:
+- job_name: 'overwritten-default'
```

Comment:
Jup, that's how we decided it back then. We cannot statically analyze this.

Not 100% a fan either, but we can always allow to drop it later should we decide that no job label is okay.